src/main/java/io/cdap/delta/bigquery/BigQueryEventConsumer.java

```diff
@@ -249,20 +249,19 @@ public synchronized void applyDDL(Sequenced<DDLEvent> sequencedEvent) throws Exc
         TableId tableId = TableId.of(project, event.getDatabase(), event.getTable());
         Table table = bigQuery.getTable(tableId);
         // TODO: check schema of table if it exists already
-        // TODO: add a way to get PK of a table through Delta API?
         if (table == null) {
           TableDefinition tableDefinition = StandardTableDefinition.newBuilder()
             .setSchema(Schemas.convert(addSequenceNumber(event.getSchema())))
             .build();
-          String description;
+          String primaryKey;
           if (event.getPrimaryKey().isEmpty()) {
-            description = "Primary Key: _sequence_num";
+            primaryKey = "_sequence_num";
           } else {
-            description = "Primary Key: " + event.getPrimaryKey().stream().collect(Collectors.joining(","));
+            primaryKey = event.getPrimaryKey().stream().collect(Collectors.joining(","));
           }
-          TableInfo tableInfo = TableInfo.newBuilder(tableId, tableDefinition)
-            .setDescription(description)
-            .build();
+          context.putState(String.format("%s-%s-primary-key", tableId.getDataset(), tableId.getTable()),
```

Comment:
since it's guaranteed that all events for this table will go to the same process, we should keep an in-memory map from tableId to primary key. Here it should write to both the in-memory map and call context.putState().

Later on, when using the primary key, we should first check the in-memory map, and only use context.getState() if it's not there. It's probably worth having some sort of PrimaryKeyStore to abstract this away.

Reply:
Sure, I can add in-memory state for primary keys.